src/test-support/mocks/src/dataplanes/_overview.ts

```diff
@@ -1,41 +1,66 @@
 import type { EndpointDependencies, MockResponder } from '@/test-support'
+
 export default ({ fake, pager, env }: EndpointDependencies): MockResponder => (req) => {
+  const query = req.url.searchParams
+  const _gateway = query.get('gateway') ?? ''
+  const _name = query.get('name') ?? ''
+  const _tags = query.get('tag') ?? ''
+
+  const inbounds = parseInt(env('KUMA_DATAPLANEINBOUND_COUNT', `${fake.number.int({ min: 1, max: 3 })}`))
   const subscriptionCount = parseInt(env('KUMA_SUBSCRIPTION_COUNT', `${fake.number.int({ min: 1, max: 10 })}`))
+  const isMtlsEnabled = env('KUMA_MTLS_ENABLED', fake.helpers.arrayElement(['false', 'true'])) === 'true'
   const { offset, total, next, pageTotal } = pager(
     env('KUMA_DATAPLANE_COUNT', `${fake.number.int({ min: 1, max: 1000 })}`),
     req,
     '/dataplanes/_overview',
   )
 
+  const tags = _tags !== ''
+    ? Object.fromEntries([_tags]
+      .filter((tag) => Boolean(tag))
+      .map(item => { const [key, ...rest] = item.split(':'); return [key, rest.join(':')] }))
+    : {}
+
+  let filterType: 'gateway_builtin' | 'gateway_delegated' | 'proxy' | undefined
+  if (_gateway === 'builtin' || _gateway === 'delegated') {
+    filterType = `gateway_${_gateway}`
+  } else if (_gateway === 'false') {
+    filterType = 'proxy'
+  } else if (tags['kuma.io/service']) {
+    if (tags['kuma.io/service'].includes('gateway_builtin')) {
+      filterType = 'gateway_builtin'
+    } else if (tags['kuma.io/service'].includes('gateway_delegated')) {
+      filterType = 'gateway_delegated'
+    } else {
+      filterType = 'proxy'
+    }
+  }
+
   return {
     headers: {},
     body: {
       total,
       items: Array.from({ length: pageTotal }).map((_, i) => {
         const id = offset + i
+
+        const isMultizone = true && fake.datatype.boolean()
+
+        const type = filterType ?? fake.helpers.arrayElement(['gateway_builtin', 'gateway_delegated', 'proxy'])
+        const mesh = `${fake.hacker.noun()}-${id}`
+        const name = `${_name || fake.kuma.dataPlaneProxyName()}-${type}-${id}`
+        const service = tags['kuma.io/service']
+
         return {
           type: 'DataplaneOverview',
-          mesh: `${fake.hacker.noun()}-${id}`,
-          name: `${fake.kuma.dataPlaneProxyName()}-${id}`,
+          mesh,
+          name,
           creationTime: '2021-02-17T08:33:36.442044+01:00',
           modificationTime: '2021-02-17T08:33:36.442044+01:00',
           dataplane: {
-            networking: {
-              address: '127.0.0.1',
-              inbound: [
-                fake.kuma.inbound(fake.kuma.serviceName()),
-              ],
-              outbound: [
-                {
-                  port: 10001,
-                  tags: {
-                    'kuma.io/service': 'frontend',
-                  },
-                },
-              ],
-            },
+            networking: fake.kuma.dataplaneNetworking({ type, inbounds, isMultizone, service }),
```

Comment:
Again, just a mental note and no action here.

---

I get the reasoning why this is being done, but it does move away from the point of our mocks being as they are. 

The idea is they should look like the API response, and in making reusable functions like this we hide the structure of the JSON away, and then the more that functions depend on each other it gets even more difficult to follow what is going on. In the end if we keep going like this it leads to:

```js
export default ({ fake, pager, env }: EndpointDependencies): MockResponder => (req) => {
   return fake.kuma.dataplane({ fake, pager, env, req }) <== the start of deeply nested functions
}
```

And then trying to see the complete JSON involves figuring out all the nested function calls.

So in removing duplication, we are trading that for less clarity. Its kinda the "small functions considered harmful" thing again.

Ideally, I'd like our mocks to be generated from the OpenAPI schemas, that would keep things clear and duplication would cease to be an issue.

I had a few other ideas on this, but I think it would be better to just wait and get on the OpenAPI auto generation thing when we can.

---

I also think adding the types to all these is just moving the problem of potentially wrong API responses over to potentially wrong types. Our types aren't what the API produces, and we've seen this issue many times, the copy/pasted API responses are.

Having or being able to copy/paste API responses straight from the actual API response of the backend with the odd primitive value replaced `fake` kinda prevents the lack of clarity, its also kind of a poor mans auto-generation.

I think overall its a bit of a balance, the sooner we can get to auto generate these and also the types themselves the better.